CRM/Report/Form/Member/Detail.php

```diff
@@ -521,6 +521,9 @@ public function contributionTableJoin() {
       WHERE cc.is_test = 0
       $dateWhere
       GROUP BY cmp.membership_id";
+    if (CRM_Utils_SQL::supportsFullGroupBy()) {
+      $sql .= ', cc.contact_id';
```

Comment:
If this is correct when full group by is enforced (supports is a misnomer I believe - all mysql supports Full Group by -then it is always correct. The point of the enforceFullGroupBy mode is to enforce reliable mysql statements, not to 'support' them in some way

Reply:
Well i can see both sides of the argument, in one sense it is correct for those systems that have ONLY_FULL_GROUP_BY on by default and therefore we are "supporting" them. We are not trying to enforce the ONLY_FULL_GROUP_BY query standard for those systems that don't have it on by default / not enforced

Reply:
Right but the only reason for the ONLY_FULL_GROUP_BY is to force application writers to write accurate mysql. All mysql supports accurate mysql.

Reply:
But if you install mysql 5.5 you don't have ONLY_FULL_GROUP_BY enabled by default, If you enable it it does something quiet different to mysql5.7's ONLY_FULL_GROUP_BY which is enabled by default and Maria10.1 and Maria10.2 have it and do different things to mysql's ONLY_FULL_GROUP_BY and in both cases aren't enabled by default. The CRM_Utils_SQL function works out if we need to re-write queries to handle the fact we maybe working in an only_full_group_by state

Reply:
OK - so you are saying there is more to it than my understanding. Here is what I have

"ONLY_FULL_GROUP_BY
Reject queries for which the select list, HAVING condition, or ORDER BY list refer to nonaggregated columns that are neither named in the GROUP BY clause nor are functionally dependent on (uniquely determined by) GROUP BY columns."

In other words - it is ALWAYS unreliable to mix nonaggregated columns with group by - but unless you have have "ONLY_FULL_GROUP_BY" enabled you can 'get away with' those unreliable queries

Reply:
Yes so what i have found is that if you enable ONLY_FULL_GROUP_BY on a Mysql 5.5 instance it is incredibly harsh and hits virtually every frecken query, as was evidenced when i tried through a PR and i made Jenkins crap its self. However on MySQL5.7 it seems more selective in someways in how it determines the queries to hit. Also if you do a plain install of MySQL5.5 you have no sqlModes set by default MySQL5.7 on a plain still will have ONLY_FULL_GROUP_BY along with NO_ZERO_DATE and a few other sqlModes set by default. MairaDB's implementation of ONLY_FULL_GROUP_BY seems similar to that of MySQL5.5 in its harshness but also not on automatically for either Maira5.5, Maria10.1, Maria10.2

Reply:
OK - but our goal is really 'to ensure all queries are valid enough to pass the mysql 5.7 interpretation' ?

Reply:
The Approach as far as I can tell has been to alter queries / rewrite queries on the fly to make them work on 5.7 in some cases in other its to re-write them such that they work for 5.7. If you look at this function https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L4680 that is an example of re-writing on the fly to try and handle ONLY_FULL_GROUP_BY by using https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_any-value but that has now run aground because of this issue https://jira.mariadb.org/browse/MDEV-10426